### PR TITLE
An ANTLR4 grammar for HIVE

### PR DESCRIPTION
**Pull request by fvoidCN**
An ANTLR4 grammar for HIVE based on version 3.1.2 of official HIVE grammar.The vast majority of grammar come from official HIVE grammar of Apache HIVE.I just translation to ANTLR4 and modify them to adapt to ANTLR4.

### CODE REVIEW

File: sql/hive/grammar/FromClauseParser.g4 [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    |
+    joinSource
+    ;
+
+
+atomjoinSource
+    :  tableSource (lateralView)*
+    |  virtualTableSource (lateralView)*
+    |  subQuerySource (lateralView)*
+    |  partitionedTableFunction (lateralView)*
```
Comment (KvanTTT) [inline sql/hive/grammar/FromClauseParser.g4:64]: Please remove excess parentheses:

```suggestion
    :  tableSource lateralView*
    |  virtualTableSource lateralView*
    |  subQuerySource lateralView*
    |  partitionedTableFunction lateralView*
```

File: sql/hive/grammar/FromClauseParser.g4 [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+tableAllColumns
+    : STAR
+    | tableName DOT STAR
+    ;
+
+// (table|column)
+tableOrColumn
+    :
+    identifier
+    ;
```
Comment (KvanTTT) [inline sql/hive/grammar/FromClauseParser.g4:33]: Please reformat in the following way:

```suggestion
tableOrColumn
    : identifier
    ;
```

For this and other rules.

File: sql/hive/grammar/FromClauseParser.g4 [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+joinSource
+    :
+    atomjoinSource (joinToken joinSourcePart (KW_ON expression {$joinToken.start.getType() != COMMA}? | KW_USING columnParenthesesList {$joinToken.start.getType() != COMMA}?)?)*
+    ;
+
+joinSourcePart
+    :
+    (tableSource | virtualTableSource | subQuerySource | partitionedTableFunction) (lateralView)*
```
Comment (KvanTTT) [inline sql/hive/grammar/FromClauseParser.g4:75]: Excess parentheses:

```suggestion
    (tableSource | virtualTableSource | subQuerySource | partitionedTableFunction) lateralView*
```

**Comment by fvoidCN** [comment]
ok, I review my code

**Comment by fvoidCN** [comment]
I had formatted this grammar.

**Pull request merged**